### ISSUE TITLE
[Docs] Add guidance for structured logging and plotting in data hyper-cleaning example

### ISSUE THREAD

**Issue by 6666-svg** [callous-youth/BOAT#29]
### Required prerequisites

- [x] I have searched the [Issue Tracker](https://github.com/callous-youth/BOAT/issues) to make sure this feature request has not already been submitted.
- [x] I have checked the documentation <https://boat.readthedocs.io/en/main/> to confirm this feature is not already supported.

### Motivation

I am running the data hyper-cleaning example in `examples/data_hyper_cleaning` and comparing different `gm_op`, `na_op`, and `fo_op` combinations.

The script prints useful metrics such as `x_itr`, `acc`, `precision`, `recall`, `F1 score`, and `val_loss`. However, for beginners who want to compare multiple operator combinations, it is not very clear how to save these metrics into structured files or plot convergence curves.

In my experiment, I had to manually save terminal outputs into log files and write an additional script to parse the results and generate CSV files and convergence curves. It would be helpful if the example or documentation provided a simple workflow for this.

### Proposed solution

I suggest adding a short section or utility script for the data hyper-cleaning example, including:

1. A minimal command-line example for running several `gm_op`, `na_op`, and `fo_op` combinations.
2. A simple way to save `x_itr`, `acc`, `F1 score`, and `val_loss` into a CSV file.
3. A plotting example for accuracy, F1 score, and validation loss convergence curves.
4. A short note for Windows users that PowerShell `Tee-Object` may save logs with a different encoding, so structured CSV logging would be more robust.

For example, the documentation could show commands like:

```powershell
python data_hyper_cleaning.py --gm_op NGD --na_op CG 2>&1 | Tee-Object -FilePath logs\ngd_cg.txt
python data_hyper_cleaning.py --gm_op DI,NGD --na_op RAD 2>&1 | Tee-Object -FilePath logs\di_ngd_rad.txt
python data_hyper_cleaning.py --fo_op MESO 2>&1 | Tee-Object -FilePath logs\meso.txt
```

### Alternatives considered

An alternative is to let users manually redirect stdout and write their own parsing scripts. This works, but it may be confusing for new users because the printed metric format is not structured as CSV or JSON.

Another alternative is to modify `data_hyper_cleaning.py` locally to save metrics during each iteration. However, a built-in or documented example would make the workflow easier to reproduce.

### Additional context

In my local run, I compared several combinations such as `NGD/CG`, `NGD/RAD`, `DI+NGD/RAD`, `GDA+NGD/CG`, `DM+NGD/RAD`, `VSO`, `VFO`, `MESO`, and `PGDO`.

The final outputs were saved as:
- `all_iterations_results.csv`
- `final_results.csv`
- `accuracy_curve.png`
- `val_loss_curve.png`
- `f1_curve.png`

This kind of output is useful for course assignments and for understanding the differences between BOAT operator combinations.
